src/tcgwars/logic/impl/gen3/RubySapphireNG.groovy

```diff
@@ -277,14 +300,20 @@ public enum RubySapphireNG implements LogicCardInfo {
           attackRequirement {}
           onAttack {
             damage 20
+            if (opp.bench) damage 10, opp.bench.select()
           }
         }
         move "Fire Spin", {
           text "100 damage. Discard 2 basic Energy cards attached to Camerupt or this attack does nothing."
           energyCost R, R, C, C
-          attackRequirement {}
+          attackRequirement {
+            assert self.cards.filterByType(BASIC_ENERGY).size() >= 2 : "$self needs 2 or more Basic Energies attached"
+          }
           onAttack {
             damage 100
+            afterDamage {
+              self.cards.filterByType(BASIC_ENERGY).select(count: 2, "Select 2 Basic Energies to discard.").discard()
```

Comment:
That sounds good. Will look into it later.

Would probably leave it as a more energy-specific static, potentially returning a boolean so it can be used as the main condition (thinking of whenever "using an attack that does nothing" is properly implemented):

```
if (forcedDiscardEnergy(amount, isBasic, energyTypeFilter)){
    damage 100
}
```

Reply:
I'll take your word on it since AFAIK being unable to discard energy causing an attack to fail is a retro-only issue.